src/Definition/GenericStorage.php

```diff
@@ -92,20 +92,17 @@ public function getBasetype()
                 return "TIMESTAMP";
             }
             return "DATE";
-        }
-        if (stristr($this->type, "int")) {
+        } else if (stristr($this->type, "int")) {
             return "INTEGER";
-        }
-        if (stristr($this->type, "float") || stristr($this->type, "double") || stristr($this->type, "real")) {
+        } else if (stripos($this->type, "float") === 0 || stripos($this->type, "real") === 0) {
             return "FLOAT";
-        }
-        if (stristr($this->type, "timestamp")) {
+        } else if (stristr($this->type, "timestamp")) {
             return "TIMESTAMP";
-        }
-        if (stristr($this->type, "bool")) {
+        } else if (stripos($this->type, "bool") === 0) {
             return "BOOLEAN";
-        }
-        if (stristr($this->type, "decimal") || stristr($this->type, "num")) {
+        } else if (stripos($this->type, "decimal") === 0 ||
+            stripos($this->type, "num") === 0 ||
+            stristr($this->type, "double")) {
```

Comment:
to prevent further confusion and random bugs, i think strict (non-substr) checking might be better. or why do you need to check `stripos($this->type, "decimal") === 0` instead of `$this->type === "decimal"?

Reply:
the test for this function does not test any partial matches (except for `double precision`)

Reply:
imho for some types it's necessary.  for example to catch `tinyInt` as an `int` it needs to stristr it.

Reply:
in the case of `num` that will catch `number`, `numeric`, `numeral`.

Reply:
if a new datatype shows up, it will fallback to `STRING` and will need to be manually added to this lib. but i think collecting a list of all datatypes from all supported DBs will not be such a PITA, can help you with that.

Reply:
@Halama removing all the automatic detections is the way to go (having each source provide its own mapping), but right now it is being used, so I think it should first be fixed like @ondrejhlavacek suggests to remove bugs and then work on removing the dependency.

Reply:
i think we can collect DBs datatypes just for this method, that will be simple

Reply:
ok.  I'll start the gathering :-)